src/status/wallet.nim

```diff
@@ -187,3 +187,21 @@ proc validateMnemonic*(self: WalletModel, mnemonic: string): string =
   result = status_wallet.validateMnemonic(mnemonic).parseJSON()["error"].getStr
 
 proc getAllCollectibles*(self: WalletModel, address: string): seq[Collectible] = getAllCollectibles(address)
+
+proc getGasPricePredictions*(self: WalletModel): GasPricePrediction =
+  try:
+    let url: string = fmt"https://etherchain.org/api/gasPriceOracle"
+    let client = newHttpClient()
+    client.headers = newHttpHeaders({ "Content-Type": "application/json" })
+    let response = client.request(url)
+    let prediction = parseJson(response.body)
+    result = GasPricePrediction(
+      safeLow: prediction["safeLow"].getStr(),
+      standard: prediction["standard"].getStr(),
+      fast: prediction["fast"].getStr(),
+      fastest: prediction["fastest"].getStr()
+    )
```

Comment:
Ideally, though this can be done possibly later, it would be prudent to cache this for a few mins to prevent excessive HTTP requests? Could use `CachedValues` like in the `BalanceManager` (https://github.com/status-im/nim-status-client/blob/master/src/status/wallet/balance_manager.nim)

Reply:
Is there any way to get gas prices for the specific network? For example, we might be on testnet (ropsten), so the gas prices would be a bit different to mainnet.  Current network can be retrieved using `settings.gretCurrentNetwork()`, with values being `Network.Mainnet` or `Network.Testnet` (for now).

Reply:
Good question, not sure etherchain supports different networks in their API. To be fair though, I believe we won't be using them forever anyways..